### PR TITLE
feat(backup): add copy actions to backup plan rules

### LINKED ISSUE #22173: Backup: Support for BackupPlanRule Copy Actions
### Describe the feature

The Level 2 BackupPlanRule doesn't support copy actions to copy backups to another vault. The Level 1 CfnBackupPlan does support this. 

### Use Case

One of the key features of AWS Backup is being able to copy backups to another Vault such as in another region or account. This is done through the use of copy actions, which today cannot be done with the L2 construct.

### Proposed Solution

Ideally it should be possible to add copy actions to a rule similar to how it would be done when creating a CfnBackupPlan:

```typescript
declare const plan: backup.BackupPlan;
plan.addRule(new backup.BackupPlanRule({
    copyActions: [{
        destinationBackupVault: myDestinationVault,

        // the properties below are optional
        lifecycle: {
          deleteAfterDays: 123,
          moveToColdStorageAfterDays: 123,
        },
      }]
...
}));
```

### Other Information

_No response_

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.42.1

### Environment details (OS name and version, etc.)

Any OS

### PR DESCRIPTION
**Pull request by rdbatch**
This PR adds the ability to specify copy actions on backup plan rules, to copy recovery points to a different vault. 

```ts
declare const plan: backup.BackupPlan;
declare const secondaryVault: backup.BackupVault
plan.addRule(new backup.BackupPlanRule({
  copyActions: [{
    destinationBackupVault: secondaryVault,
    moveToColdStorageAfter: Duration.days(30),
    deleteAfter: Duration.days(120),
  }]
}));
```

The naming and types for `moveToColdStorageAfter` and `deleteAfter` are consistent with the parent `Rule` type to avoid confusion.

This closes #22173.

Note: While working on this, I discovered that there's a gap in the validation for `Rule` where if both `moveToColdStorageAfter` and `deleteAfter` are specified, then `deleteAfter` must be at least 90 days later than the `moveToColdStorageAfter` point. I've added validation for this in the `copyActions` validation, but I think fixing this for the rest of `Rule` probably warrants a separate bug fix issue and PR.

----

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [x] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [x] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-backup/README.md [hunk 1]
```diff
@@ -99,6 +99,21 @@ plan.addRule(new backup.BackupPlanRule({
 }));
 ```
 
+Rules can also specify to copy recovery points to another Backup Vault using `copyActions`. Copied recovery points can
+optionally have `moveToColdStorage` and `deleteAfter` configured.
+
+```ts
+declare const plan: backup.BackupPlan;
+declare const secondaryVault: backup.BackupVault
```
Comment (kaizencc) [inline packages/@aws-cdk/aws-backup/README.md:107]:
```suggestion
declare const secondaryVault: backup.BackupVault;
```

File: packages/@aws-cdk/aws-backup/lib/rule.ts [hunk 2]
```diff
@@ -185,6 +216,20 @@ export class BackupPlanRule {
       throw new Error(`'deleteAfter' must be between 1 and 35 days if 'enableContinuousBackup' is enabled, but got ${props.deleteAfter.toHumanString()}`);
     }
 
+    if (Array.isArray(props.copyActions) && props.copyActions.length > 0) {
```
Comment (kaizencc) [inline packages/@aws-cdk/aws-backup/lib/rule.ts:219]:
```suggestion
    if (props.copyActions && props.copyActions.length > 0) {
```

nit: not that you're wrong, but this is _maybe_ simpler?

File: packages/@aws-cdk/aws-backup/lib/rule.ts [hunk 3]
```diff
@@ -185,6 +216,20 @@ export class BackupPlanRule {
...
+      props.copyActions.forEach(copyAction => {
+        if (copyAction.deleteAfter && copyAction.moveToColdStorageAfter &&
+          copyAction.deleteAfter.toDays() < copyAction.moveToColdStorageAfter.toDays()) {
+          throw new Error('`deleteAfter` in copyActions must be greater than corresponding `moveToColdStorageAfter`');
+        }
+
+        if (copyAction.deleteAfter && copyAction.moveToColdStorageAfter &&
+          copyAction.deleteAfter.toDays() < copyAction.moveToColdStorageAfter.toDays() + 90) {
+          throw new Error('`deleteAfter` in copyActions must at least 90 days later than corresponding `moveToColdStorageAfter`');
+        }
```
Comment (kaizencc) [inline packages/@aws-cdk/aws-backup/lib/rule.ts:229]:
```suggestion
        if (
          copyAction.deleteAfter && !Token.isUnresolved(copyAction.deleteAfter)
          copyAction.moveToColdStorageAfter && !Token.isUnresolved(copyAction.moveToColdStorageAfter)
          copyAction.deleteAfter.toDays() < copyAction.moveToColdStorageAfter.toDays() + 90,
        ) {
          throw new Error([
            '\'deleteAfter\' must at least 90 days later than corresponding \'moveToColdStorageAfter\'',
            `received 'deleteAfter: ${copyAction.deleteAfter.toDays()}' and 'moveToColdStorageAfter: ${copyAction.moveToColdStorageAfter.toDays()}'`,
          ].join('\n'));
        }
```

File: packages/@aws-cdk/aws-backup/lib/rule.ts [hunk 4]
```diff
@@ -70,6 +70,37 @@ export interface BackupPlanRuleProps {
...
+   */
+  readonly destinationBackupVault: IBackupVault;
+
+  /**
+   * Specifies the duration after creation that a copied recovery point is deleted from the destination vault.
+   * Must be at least 90 days greater than `moveToColdStorageAfter`, if specified.
+   *
+   * @default - recovery point is never deleted
+   */
+  readonly deleteAfter?: Duration;
```
Comment (kaizencc) [inline packages/@aws-cdk/aws-backup/lib/rule.ts:97]:
```suggestion
   */
 
  readonly deleteAfter?: Duration;
```

**Review by kaizencc — CHANGES_REQUESTED** [review]
This looks great @rdbatch02! Just a few minor comments.

File: packages/@aws-cdk/aws-backup/lib/rule.ts (hunk 2 above)
Comment (rdbatch02) [reply packages/@aws-cdk/aws-backup/lib/rule.ts]: Good call, I'm all for making this simpler.

File: packages/@aws-cdk/aws-backup/lib/rule.ts (hunk 3 above)
Comment (rdbatch02) [reply packages/@aws-cdk/aws-backup/lib/rule.ts]: Thanks, this is much more concise now and having the additional feedback in the error message should be very helpful. I've updated this and the corresponding tests.

File: packages/@aws-cdk/aws-backup/README.md [hunk 5]
```diff
@@ -99,6 +99,21 @@ plan.addRule(new backup.BackupPlanRule({
 }));
 ```
 
+Rules can also specify to copy recovery points to another Backup Vault using `copyActions`. Copied recovery points can
+optionally have `moveToColdStorage` and `deleteAfter` configured.
```
Comment (kaizencc) [inline packages/@aws-cdk/aws-backup/README.md:103]:
```suggestion
optionally have `moveToColdStorageAfter` and `deleteAfter` configured.
```

**Review by kaizencc — APPROVED** [review]
Thanks @rdbatch02!

**Pull request merged**